Game.cs

```diff
@@ -40,41 +41,177 @@ public Player CurrentPlayer
 
     public int CurrentPlayerIndex { get; private set; }
 
+    public static Game LoadFromFile(string filename)
+    {
+        return FromJSON(File.ReadAllText(filename));
+    }
+
+    public static Game FromJSON(string input)
+    {
+        var data = JObject.Parse(input);
+        var result = new Game();
+        result.CurrentPlayerIndex = data.GetValue("current").ToObject<int>();
+        result.Ticks = data.GetValue("ticks").ToObject<int>();
+
+        result.Finished = data.GetValue("finished").ToObject<bool>();
+        foreach (JObject playerJSON in (JArray)data.GetValue("players"))
+        {
+            var player = new Player(result.Manager, playerJSON.GetValue("color").ToObject<Color>());
+            player.Resources.Add(playerJSON.GetValue("resources").ToObject<ResourceBag>());
+            foreach (JObject armyJSON in (JArray)playerJSON.GetValue("armies"))
+            {
+                var army = armyJSON.GetValue("army").ToObject<Army>();
+                player.AddArmy(army, armyJSON.GetValue("origin").ToObject<Pos>());
+                var dest = armyJSON.GetValue("destination");
+                if (dest != null)
+                {
+                    result.Manager.MoveArmy(army, dest.ToObject<Pos>());
+                }
+            }
+
+            result.Players.Add(player);
+        }
+
+        var ownersArray = (JArray)data.GetValue("owners");
+
+        for (int x = 0; x < World.WIDTH; x++)
+        {
+            for (int y = 0; y < World.HEIGHT; y++)
+            {
+                var playerNumber = ownersArray[(x * World.HEIGHT) + y].ToObject<int>();

```

Comment:
`x * World.HEIGHT + y` is also good because multiply operation has a higher priority 😄